src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -64,6 +65,54 @@ std::vector<std::string> get_supported_metrics() {
           "confusion_matrix", "roc_curve"};
 }
 
+size_t count_correct_predictions(size_t prediction_window, size_t num_classes, const shared_float_array& output_chunk, 
+  const shared_float_array& label_chunk, data_iterator::batch::chunk_info info) {
```

Comment:
We're only using `info` to get `info.num_samples`, so you just pass in `num_samples` instead. Furthermore, we're only using `num_samples` to count how many times to go through the inner `while` loop. That number must be equal to the `num_predictions` value you're computing anyway in `get_accuracy_per_batch`. So could we just pass in `num_predictions` instead?

Nit: let's indent this line with four spaces instead of two, to make it more visually obvious that this is a line continuation, and not e.g. the first line of the function body